### ISSUE TITLE
mqttapi.listen_event() - Async implemented properly?

### ISSUE THREAD

**Issue by rr326** [AppDaemon/appdaemon#1085]
I don't understand asyncio very well so this may be totally fine, but I think it is wrong.

In mqttapi.py:
```python
    @utils.sync_wrapper
    async def listen_event(self, callback, event=None, **kwargs):
    # ...
    return super(Mqtt, self).listen_event(callback, event, **kwargs)
```
**Should it be?**
```python
return await super(...) 
```

I've implemented my own subclass on Mqtt and I tried to build a proper sync/async listen event. But I'm getting
an asyncio Task returned, instead of the value itself.

```python
cancel_handle = await self.listen_event(...)

assert isinstance(cancel_handle, asyncio.Task) # True
```

I think mqttapi.listen_event should `return await super()...`

Apologies if this is my misunderstanding.

**Comment by Odianosen25** [comment]
Hello @rr326,

Yes you are right there should be an “await” there. Can you be kind to submit a PR to correct this pls?

Regards

**Comment by rr326** [comment]
You bet...

**Comment by rr326** [comment]
Fixed in #1086

**Comment by Odianosen25** [comment]
Thanks @rr326
